generators/server/templates/src/main/docker/jhipster-control-center.yml.ejs

```diff
@@ -0,0 +1,67 @@
+# To reach the application from a browser, you need to add '127.0.0.1 host.docker.internal' to your hosts file.
+### Discovery mode
+# JHCC support 3 kinds of discovery mode: Consul, Eureka and static
+# In order to use one, please set SPRING_PROFILES_ACTIVE to one (and only one) of this values: consul,eureka,static
+### Discovery properties
+# According to the discovery mode choose as Spring profile, you have to set the right properties
+# please note that current properties are set to run JHCC with default values, personalize them if needed
+# and remove those from other modes. You can only have one mode active.
+#### Eureka
+#       - EUREKA_CLIENT_SERVICE_URL_DEFAULTZONE=http://admin:admin@host.docker.internal:8761/eureka/
+#### Consul
+#       - SPRING_CLOUD_CONSUL_HOST=host.docker.internal
+#       - SPRING_CLOUD_CONSUL_PORT=8500
+#### Static
+# Add instances to "MyApp"
+#       - SPRING_CLOUD_DISCOVERY_CLIENT_SIMPLE_INSTANCES_MYAPP_0_URI=http://host.docker.internal:8081
+#       - SPRING_CLOUD_DISCOVERY_CLIENT_SIMPLE_INSTANCES_MYAPP_1_URI=http://host.docker.internal:8082
+# Or add a new application named MyNewApp
+#       - SPRING_CLOUD_DISCOVERY_CLIENT_SIMPLE_INSTANCES_MYNEWAPP_0_URI=http://host.docker.internal:8080
+<%_ let discoveryProfile = serviceDiscoveryType ? serviceDiscoveryType : 'static'; _%>
+version: '<%= DOCKER_COMPOSE_FORMAT_VERSION %>'
+services:
+  jhipster-control-center:
+    image: '<%= DOCKER_JHIPSTER_CONTROL_CENTER %>'
+    environment:
+      - _JAVA_OPTIONS=-Xmx512m -Xms256m
+      - SPRING_PROFILES_ACTIVE=prod,swagger,<%= discoveryProfile %><% if (authenticationType === 'oauth2') { %>,oauth2<% } %>
+      - JHIPSTER_SLEEP=30 # gives time for other services to boot before the application
+      <%_ if (authenticationType === 'jwt') { _%>
+      - SPRING_SECURITY_USER_PASSWORD=admin
+      - JHIPSTER_SECURITY_AUTHENTICATION_JWT_BASE64_SECRET=<%= jwtSecretKey %>
+      <%_ } else if (authenticationType === 'oauth2') { _%>
+      # For keycloak to work, you need to add '127.0.0.1 keycloak' to your hosts file
+      - SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_OIDC_ISSUER_URI=http://keycloak:9080/auth/realms/jhipster
+      - SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_ID=jhipster-control-center
+      - SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET=jhipster-control-center
+      <%_ } _%>
+      <%_ if (discoveryProfile === 'eureka') { _%>
+      - EUREKA_CLIENT_SERVICE_URL_DEFAULTZONE=http://admin:admin@host.docker.internal:8761/eureka/
+      <%_ } else if (discoveryProfile === 'consul') { _%>
+      - SPRING_CLOUD_CONSUL_HOST=host.docker.internal
+      - SPRING_CLOUD_CONSUL_PORT=8500
+      <%_ } else { _%>
+      - SPRING_CLOUD_DISCOVERY_CLIENT_SIMPLE_INSTANCES_<%= baseName.toUpperCase() %>_0_URI=http://host.docker.internal:<%= serverPort %>
+      <%_ } _%>
+    ports:
+      - 7419:7419
```

Comment:
@avdev4j To make the `host.docker.internal` host resolution work on Linux, here is the magic line you need to add to the docker-compose.

```
    command:
      - /bin/sh
      - -c
      # Patch /etc/hosts to support resolving host.docker.internal to the internal IP address used by the host in all OSes
      - echo "`ip route | grep default | cut -d ' ' -f3` host.docker.internal" | tee -a /etc/hosts > /dev/null && java -jar /jhipster-control-center.jar
```

I have installed the iproute command in the docker image in this jhcc PR which is already merged and deployed to docker hub: https://github.com/jhipster/jhipster-control-center/pull/85